pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarStreamMetadataProvider.java

```diff
@@ -124,19 +127,28 @@ public StreamPartitionMsgOffset fetchStreamPartitionOffset(OffsetCriteria offset
 
         for (int p = newPartitionStartIndex; p < partitionedTopicNameList.size(); p++) {
 
-          Reader reader =
-              _pulsarClient.newReader().topic(getPartitionedTopicName(p)).startMessageId(_config.getInitialMessageId())
-                  .create();
+          Consumer consumer = _pulsarClient.newConsumer().topic(getPartitionedTopicName(p))
+              .subscriptionInitialPosition(_config.getInitialSubscriberPosition())
+              .subscriptionName(ConsumerName.generateRandomName()).subscribe();
 
-          if (reader.hasMessageAvailable()) {
-            Message message = reader.readNext();
+          Message message = consumer.receive(timeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
maybe add a comment here, that the offsetCriteria from StreamConfig param supplied to this method is ignored on purpose when deciding the offset, and that purposely "earliest" is used for all ew partitions

Reply:
honoring stream config param now instead of global config.